### PR TITLE
fix: iOS 18 Toggle tap gesture

### PR DESCRIPTION
**Pull request by rnr**
In iOS 18, parental view blocks the tap gesture for Toogle, and tapping only works in a small area above Toggle. Drag action doesn't work too. Added priority for the Toggle gestures (both tap and drag).

| Before | After |
| ------ | ------ |
| <video src="https://github.com/user-attachments/assets/bfe2f698-2c5b-4923-860e-53fc666e6fcb"> | <video src="https://github.com/user-attachments/assets/f92a98ac-ee23-43f4-a48d-0464dc58b1bb"> |

### CODE REVIEW

**Comment by shafqat-muneer** [comment]
@rnr  Based on my observations, this issue seems to be specific to the simulator and occurs when the trackpad is clicked on a Mac. If the user uses a tap gesture on the trackpad, the issue does not appear, even in the simulator. 

It's great that this scenario has been addressed. However, there are two additional places where the `Toggle` control is being used. Could you kindly address it there as well?
```
- CourseDatesView
- DatesAndCalendarView
```

**Comment by rnr** [comment]
> @rnr Based on my observations, this issue seems to be specific to the simulator and occurs when the trackpad is clicked on a Mac. If the user uses a tap gesture on the trackpad, the issue does not appear, even in the simulator.
> 
> It's great that this scenario has been addressed. However, there are two additional places where the `Toggle` control is being used. Could you kindly address it there as well?
> 
> ```
> - CourseDatesView
> - DatesAndCalendarView
> ```

@shafqat-muneer 
I have this issue on my iPhone with iOS 18.1 when build from 2U/develop right now - so it's not Simulator or trackpad bug.
And you are right `CourseDatesView` has this bug too since .onTapGesture is using there. Let me fix it. Thank you

**Comment by shafqat-muneer** [comment]
> @shafqat-muneer I have this issue on my iPhone with iOS 18.1 when build from 2U/develop right now - so it's not Simulator or trackpad bug. And you are right `CourseDatesView` has this bug too since .onTapGesture is using there. Let me fix it. Thank you

Got it. It's great that the issue has been identified and fixed 🎉. I tested it on a device running OS 17.3 but will also check on the latest OS version.

File: Course/Course/Presentation/Subviews/CourseVideoDownloadBarView/CourseVideoDownloadBarView.swift [hunk 1]
```diff
@@ -135,13 +135,15 @@ struct CourseVideoDownloadBarView: View {
         Toggle("", isOn: .constant(viewModel.isOn))
```
Comment (forgotvas) [inline Course/Course/Presentation/Subviews/CourseVideoDownloadBarView/CourseVideoDownloadBarView.swift:135]: @rnr did you try to refactor that code and use @Published property here...
so in view model:
```swift
@Published var isOn:Bool = false{
  didSet {
      if !isInternetAvaliable {
         onNotInternetAvaliable?()
         return
     }
     Task { await onToggle()  }
  }
}
```

and in view:
```swift
Toggle("", isOn: $viewModel.isOn)
  .toggleStyle(SwitchToggleStyle(tint: Theme.Colors.toggleSwitchColor))
  .padding(.trailing, 15)
  .accessibilityIdentifier("download_toggle")
```

Is there any reason why we are using TapGesture here?

File: Course/Course/Presentation/Dates/CourseDatesView.swift [hunk 2]
```diff
@@ -462,9 +462,16 @@ struct CalendarSyncView: View {
                 Toggle("", isOn: .constant(viewModel.isOn))
```
Comment (forgotvas) [inline Course/Course/Presentation/Dates/CourseDatesView.swift:462]: Same here

File: Course/Course/Presentation/Subviews/CourseVideoDownloadBarView/CourseVideoDownloadBarView.swift (hunk 1 above)
Comment (rnr) [reply Course/Course/Presentation/Subviews/CourseVideoDownloadBarView/CourseVideoDownloadBarView.swift:135]: @forgotvas  The reason is that Toggle is not used here as a classic Toggle. The user interacts with the Toggle and we show an Alert where we have to decide whether the Toggle should change its state or not.

Comment (forgotvas) [reply Course/Course/Presentation/Subviews/CourseVideoDownloadBarView/CourseVideoDownloadBarView.swift:135]: got it

**Review by forgotvas — APPROVED** [review]

**Comment by rnr** [comment]
@shafqat-muneer 
PR is ready for another pass.
Thank you

**Review by shafqat-muneer — APPROVED** [review]
LGTM 👍

**Pull request merged**